Fix unchecked struct with custom constructor

sorawee opened the pull request:
Fix a bug introduced in #3267

Related to: #3300

jbclements commented:
@sorawee can you confirm that this was a simple typo? That is, the changed "make-struct-info" would never be correct, and this was just a case of a small piece of untested code? I'm super-excited about getting this merged.

sorawee commented:
It's not a typo per se. Before #3267, the `make-struct-info` call is correct. But in #3267, I added one more argument to the call. This simultaneously affects the functions:
- `make-self-ctor-checked-struct-info`
- `make-checked-struct-info`
- `make-self-ctor-struct-info`
- `make-struct-auto-info`
- and `make-struct-info`

However, I also changed every function mentioned above except the last one (which I forgot) so that they accept one more argument. This can be done because these functions except the last one are not exported.

This PR fixes the mistake in #3267 by changing `make-struct-info` to `make-field-struct-info`, which accepts one more argument. Note that we can't directly change `make-struct-info` like we did before because `make-struct-info` is public, so changing it would break backward compatibility.
